docs/design/dp_network_policy_arktos_mizar_int.md

```diff
@@ -0,0 +1,77 @@
+
+### Network Policy Create/Update Calculation
+
+Update_and_retrieve_endpoint_names: 
+
+1. Update operator store for label_networkpolicies_store[label] = network_policy which is stored in mizar operator store. Note that operator store is a in-memory storage system in mizar.
+2.	Based on label vs network_policy dictionary in step 1, find all the pod’s names that satisfy label key-value set. NOTE: this step involves calling Kubernetes API server to get list of pods.
+3.	From step 2, we got the list of pods that satisfy podSelector label, which are the pods that will be affected by the network policy creation/update. From the pod list, we then retrieve from mizar operator store to get list of endpoint objects associated with these pods. Note: each endpoint obj keeps track of a list of network policy associated with this endpoint. Later in the process, it needs update this list with affected network policy names. 
+4.	Add all endpoints retrieved in step 3 and add them to a list (ep_names_to_be_handled), and then remove endpoints that was in networkpolicy_endpoints_ingress_store or networkpolicy_endpoints_egress_store (this is stored in mizar operator store) but no longer in ep_names_to_be_handled list. 
+
+handle_networkpolicy_update_delete
+
+1.	It gets the list of endpoints that are affected by the network policy update or create (step 4 above), and then mizar operator tries to retrieve all the endpoint objects (stored in mizar operator store) from the list. 
+2.	In step 3 above, each affected endpoint obj keeps track of all the network policy associated with this endpoint obj. Thus, mizar operator can get a list of network policies that are associated with this endpoint object and calculate the bit map which will update ebpf map later. 
+
+## Design Consideration and Challenges  
+
+Previously, with kind cluster, mizar can call kubernete API server directly and access operator store at the same time. However, with arktos integration, arktos API server and network policy operator (in red) does not communication with each other directly. 
+
+* The communication between arktos API server and mizar network policy operator is one directional. Mizar network policy operator cannot send requests back to arktos API server. 

```

Comment:
Mizar can reply to to Arktos gRPC client, it just can't star the conversation :)